crates/rpc/rpc-types-compat/src/proof.rs

```diff
@@ -7,7 +7,7 @@ use reth_trie_common::{AccountProof, StorageProof};
 /// Creates a new rpc storage proof from a primitive storage proof type.
 pub fn from_primitive_storage_proof(proof: StorageProof) -> EIP1186StorageProof {
     EIP1186StorageProof {
-        key: JsonStorageKey::Hash(proof.key),
+        key: JsonStorageKey::Number(proof.key.into()),
```

Comment:
hmm, I think this is more nuanced than that, because the reason why this even is an enum is so we can mirror the input:

https://github.com/ethereum/go-ethereum/blob/00a73fbcce3250b87fc4160f3deddc44390848f4/internal/ethapi/api.go#L680C25-L693

so i guess we'd need to take this into account here:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/helpers/state.rs#L120-L125